Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -684,57 +689,94 @@ void ABenchmarkGymGameModeBase::OnRepTotalNPCs()
 	OnTotalNPCsUpdated(TotalNPCs);
 }
 
-int32 ABenchmarkGymGameModeBase::GetActorClassCount(TSubclassOf<AActor> ActorClass) const
+void ABenchmarkGymGameModeBase::OnActorCountReportIdx()
 {
-	const UCounterComponent* CounterComponent = Cast<UCounterComponent>(GameState->GetComponentByClass(UCounterComponent::StaticClass()));
-	check(CounterComponent != nullptr)
-
-	return CounterComponent->GetActorClassCount(ActorClass);
+	UpdateAndReportActorCounts();
 }
 
-void ABenchmarkGymGameModeBase::SetLifetime(int32 Lifetime)
+void ABenchmarkGymGameModeBase::UpdateAndReportActorCounts()
 {
-	if (TestLifetimeTimer.SetTimer(Lifetime))
+	const UWorld* World = GetWorld();
+	bool bSpatialEnabled = USpatialStatics::IsSpatialNetworkingEnabled();
+	const FString WorkerID = bSpatialEnabled ? GetGameInstance()->GetSpatialWorkerId() : TEXT("Worker1");
+	if (WorkerID.IsEmpty())
 	{
-		TestLifetimeTimer.SetLock(true);
-		UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Setting NFR test lifetime %d"), Lifetime);
+		NFR_LOG(LogBenchmarkGymGameModeBase, Error, TEXT("%s: Worker ID was empty"), *NFRFailureString);
+		return;
 	}
-	else
+
+	ActorCountMap& ThisWorkerActorCounts = WorkerActorCounts.FindOrAdd(WorkerID);
+	for (auto const& Pair : ExpectedActorCounts)
 	{
-		UE_LOG(LogBenchmarkGymGameModeBase, Warning, TEXT("Could not set NFR test liftime to %d. Timer was locked."), Lifetime);
+		const TSubclassOf<AActor>& ActorClass = Pair.Key;
+		if (!USpatialStatics::IsActorGroupOwnerForClass(World, ActorClass))
+		{
+			continue;
+		}
+
+		const FExpectedActorCountConfig& Config = Pair.Value;
+		if (Config.MinCount > 0)
+		{
+			int32 TotalCount = 0;
+			int32& AuthCount = ThisWorkerActorCounts.FindOrAdd(ActorClass);
+			GetActorCount(ActorClass, TotalCount, AuthCount);
+			NFR_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Actor count for %s: ActorClass: %s Count: %d, AuthCount: %d"), *WorkerID , *ActorClass->GetName(), TotalCount, AuthCount);
```

Comment:
WorkerID is redundant here because it'll be included in that workers logs

Reply:
Sure, although I did find it useful on the auth server, as there are a lot of actor counts being printed.

Reply:
I'm not sure I follow - doesn't this only log the local worker's actor counts?

Reply:
Yes, but on the auth server there are logs for the total actor count as well as the auth actor counts for all server worker. Just thought it made sense to be explicit there to avoid any potential confusing.